polynote-frontend/polynote/state/state_handler.ts

```diff
@@ -274,6 +291,41 @@ export class StateHandler<S> extends StateWrapper<S> {
     }
 }
 
+/**
+ * An updatable StateView for a optional state
+ */
+export class OptionalStateHandler<S> extends UpdatableState<S | undefined> {
+    constructor(view: StateView<S | undefined>, disposable: IDisposable = new Disposable()) {
+        super(view, disposable);
+    }
+
+    lens<K extends keyof S>(key: K): OptionalStateHandler<Exclude<S[K], undefined>> {
+        return this.lensOpt(key)
+    }
+
+    lensOpt<K extends keyof S>(key: K): OptionalStateHandler<Exclude<S[K], undefined>> {
+        const view = this.viewOpt<S, K>(key);
+        const lens = new OptionalStateHandler(view);
+        lens.addObserver((viewState, _, src) => {
+            if (this.state === undefined && viewState === undefined)
```

Comment:
Don't we need to propagate this `undefined` value to the lens state?

Reply:
```typescript
const state: StateHandler<Foo> = StateHandler.from({bar: undefined})
const lens = state.lensOpt("bar")
state.update(undefined);
```
In this case I don't think anyone needs to be notified of the new `undefined`?

Reply:
Hmm, I wonder if there's any opportunity for this to get out of sync. 

continuing your example: 

```typescript
lens.update("blah")
lens.update(undefined)
```

Would that second update work?

Reply:
```typescript
const state: StateHandler<Foo | undefined> = StateHandler.from({bar: undefined})
const lens = state.lensOpt("bar")
state.update(undefined); // lens should not get an update
lens.update("blah");         // both lens and state should see an update – lens has changed to "blah" and state has changed to {bar: "blah"}
lens.update(undefined);  // both lens and state should see an update – lens has changed to "undefined" and state has changed to {bar: undefined}

Reply:
👍  ok, I'm convinced!